File: backend/src/main/java/com/vichovong/restaurant_pos/feature/menu/entity/MenuItem.java
package com.vichovong.restaurant_pos.feature.menu.entity;

import com.vichovong.restaurant_pos.common.entity.BaseEntity;
import com.vichovong.restaurant_pos.feature.currency.entity.Currency;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.math.BigDecimal;

@Getter
@Setter
@NoArgsConstructor
@Entity
@Table(name = "menu_items")
public class MenuItem extends BaseEntity {

    @Column(name = "name_en", nullable = false, length = 150)
    private String nameEn;

    @Column(name = "name_km", nullable = false, length = 150)
    private String nameKm;

    @Column(name = "description_en", columnDefinition = "TEXT")
    private String descriptionEn;

    @Column(name = "description_km", columnDefinition = "TEXT")
    private String descriptionKm;

    @Column(nullable = false, precision = 12, scale = 2)
    private BigDecimal price;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "currency_id", nullable = false)
    private Currency currency;

    @Column(name = "image_url", length = 500)
    private String imageUrl;

    @Column(nullable = false)
    private boolean available = true;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "category_id", nullable = false)
    private Category category;
}

File: frontend/src/features/menu/schemas/menu-item.ts
import { z } from "zod"

export const menuItemSchema = z.object({
  nameEn: z.string().min(1, "English name is required"),
  nameKm: z.string().min(1, "Khmer name is required"),
  descriptionEn: z.string(),
  descriptionKm: z.string(),
  price: z
    .number("Price must be a number")
    .min(0, "Price can't be negative"),
  currencyCode: z.string().min(1, "Currency is required"),
  categoryId: z.string().min(1, "Category is required"),
  available: z.boolean(),
})

export type MenuItemValues = z.infer<typeof menuItemSchema>
